### ISSUE TITLE
[Contribution Idea]

### ISSUE THREAD

**Issue by wgoops** [Ajarmar/universal-pokemon-randomizer-zx#246]
**Is your idea a new setting, a change to an existing setting, or something else? Please describe clearly:** 

Enable exporting of randomized game file log as a machine-readable format (ex. json). Initial PR would be specifically for data directly related to randomization which affects pokemon (moveset changes, stat changes, location found, etc). 

**What problem would your contribution solve? Please describe:** 

Current development of third-party tools is hampered, as interfacing with randomizer changes requires either parsing log files to a machine-readable format or forking the project and making changes to source code. Allowing for randomizer data to be machine-readable eases interface development between UPR and (currently sparse) third-party tools. 

Wiki would be updated to provide a basic description/example of the json output format

**Additional context:**

In plain english, I want to play around with making a local browser-based UI for viewing randomized pokemon evolutions, move changes, etc., as navigating the log file is kinda painful. To facilitate this, I'm going to allow the randomizer to optionally output a JSON of each pokemon's moveset, evolutions, stat changes, etc. I figure if I'm already gonna play around with developing an interface for the randomizer, I might as well see if the folks maintaining this project would be interested in merging it with the master branch (as long as it isn't a buggy piece of crap ofc).

**Comment by tom-overton** [comment]
Sure, this seems like it could be a valuable feature for some. However, before you begin work on it, please post your proposed JSON schema here so we can make sure it looks good. It saves us all time to hash out the schema here rather than in the comments of a pull request.

**Comment by tom-overton** [comment]
@wgoops Are you still planning on doing this? If not, I'll close this issue for now.

**Comment by wgoops** [comment]
Nah, feel free to close it
